### PR TITLE
Fix stamp overlap - MatchTimestamp

### PR DESCRIPTION
**Pull request by marco-silveira**
Golang's ```regexp``` package, build under google's RE2 engine, doesn't support backtracking and look around assertions (https://github.com/golang/go/issues/18868). Since negative lookahead is the most straight forward solution, I've found this: https://github.com/dlclark/regexp2.

The only downside of using ```regexp2``` (other than adding one more dependency to the project) is that it doesn't have constant time guarantees like the built-in ```regexp``` package, but since it's only used to Timestamp matches, I don't see it being a problem.

https://godoc.org/github.com/dlclark/regexp2
Ref #8

### CODE REVIEW

**Comment by cyucelen** [comment]
Could we just match for `Jan _2 15:04:05(\s|$)` and get rid of extra complexity stuff if possible. We can assume that no character comes after `Stamp`. We are marking the extra space in this case but, I believe simplicity is more important in this trade-off.

**Comment by jnatalzia** [comment]
+1 to the simplicity

**Comment by marco-silveira** [comment]
> Could we just match for `Jan _2 15:04:05(\s|$)` and get rid of extra complexity stuff if possible. We can assume that no character comes after `Stamp`. We are marking the extra space in this case but, I believe simplicity is more important in this trade-off.

Sure, I think that's ok in these case too, but I don't know how to solve the overlaps with ANSIC and UnixDate layouts without using negative lookbehind. That's the main reason that made me use regexp2.

I will make the changes and update the PR, later on we can discuss the other overlaps @cyucelen

**Comment by cyucelen** [comment]
Of course @MarcoAntonioSilveira ! I feel like it will be a good challenge.

**Comment by marco-silveira** [comment]
@cyucelen :recycle:

File: go.mod [hunk 1]
```diff
@@ -3,6 +3,7 @@ module github.com/cyucelen/marker
 go 1.12
 
 require (
+	github.com/dlclark/regexp2 v1.2.0
```
Comment (jnatalzia) [inline go.mod]: do we still need these dependencies?

Comment (marco-silveira) [reply go.mod]: Ops 😅

**Comment by cyucelen** [comment]
Is it ok to merge @MarcoAntonioSilveira ?

**Comment by marco-silveira** [comment]
@cyucelen yes

**Pull request merged**